config/README.md

```diff
@@ -0,0 +1,119 @@
+# Dependencies installation
+
+For a faster creation of the environment (and Conda environments in general),
+you can also install [Mamba][mamba] on top of Conda. In that case, replace
+`conda` with `mamba` in the commands below (particularly in 
+`conda env create`).
+
+Create and activate the virtual environment with the required dependencies
+with Conda:
+
+```bash
+conda env create -f environment.yml
+conda activate mirflowz
+```
+
+If you plan to run _MIRFLOWZ_ via Conda, we recommend using the following
+command for a faster environment creation specially if you will run it on an
+HPC cluster.
+
+```bash
+conda config --set channel_priority strict
+```
+In addition, if you want to run _MIRFLOWZ_ via Singularity and do not already
+have it installed globally on your system, you must further update the Conda
+environment with:
+
+```bash
+conda env update -f environment.root.yml
+```
+
+> Mind that you must have the environment activated and root permissions on
+> your system to install Singularity.
+
+
+# Run the workflow on your own samples
+
+In order to run _MIRFLOWZ_ on your own samples, we recommend having all the
+input files inside a dedicated directory. This way is easier to keep the data
+together and reproduce an analysis. Assuming that your current directory is
+the repository's root directory, create a directory to store all your data
+and traverse to it with:
+
+```bash
+mkdir path/to/your_run
+cd path/to/your_run
+```
+
+### 1. Prepare the sample table
+
+Create an empty sample table. Refer to the
+[sample.tsv](../test/test_files/samples_table.tsv) test file to see what the
+table must look like or use it as a template.
+
+```bash
+touch samples.tsv
+```
+
+> Fill the sample table according to the following requirements:  
+>
+> - `sample`. This column contains the library name.  
+> - `sample_file`. In this column, you must provide the path to the library file.
+> The path must be relative to the working directory.  
+> - `adapter`.  This field must contain the adapter sequence in capital letters.  
+> - `format`. In this field you must state the library format. It can either be 
+> `fa` if providing a FASTA file or `fastq` if the library is a FASTQ file.  
```

Comment:
I think you don't need to write "This column", "In this field" etc. for each of these, I guess that's really obvious.

My suggestion:
- `sample`: Arbitrary name for the miRNA sequencing library
- `sample_file`: Path to the miRNA sequencing library file. <CONTINUE WITH EXPLANATION OF RELATIVE PATHS; see other comment>
- `adatper`: Sequence of the 3'-end adapter used during library preparation, in capital letters
- `format`: One of `fa` or `fastq`, if the library file is in FASTA or FASTQ format, respetively

> BTW: It's a bit weird/inconsistent that we require `fa` or `fastq` rather than `fa` and `fq`, or `fasta` and `fastq`. Probably it's even better to allow both short and long forms, and independent of case. Always nice to make it easier for the user. And while we are at it, I don't see why we can't just call an `.upper()` on the `adapter` as well, so as to allow users to provide all lower case, all upper case or a mix thereof. Bit annoying to change if they copy the adapter seq from somewhere and it's not in the case we require. Fine with me if you include these changes in _this_ PR, as it is directly related to providing easy instructions here.

Reply:
The solution I found to allow the end-user to use both the short and the long version of the FASTA and FASTQ files is to duplicate the rules that depends on it using the different versions of the file extension ( see [stackOverflow answer](https://stackoverflow.com/a/56408220)

Reply:
No. There must be a better way. It's just a string param. I'm pretty sure Snakemake allows applying functions to parameters, so you should be able to write code that converts it to whatever you need, while still allowing users to provide something more simple/intuitive/forgiving. CJ should know about this, she has done something similar for ZARP.